Scaling the Waku Protocol: A Performance Analysis with Wakurtosis

Daimakaimura opened the pull request:
Research blog post on Wakurtosis and scaling Waku.

- Updated the body of the article expanding some parts
- Added bandwidth multiplier heat-map figures
- Commented the Discv5 section as we are still waiting for some final data
- Added reference to the Tech Report addendum 
- Added results of last simulations for the Discv5 vs non-Discv5 comparison
- Added results of simulations with 0 message rate in discussion
- Updated calculations and plots
- Updated the results section and plots with the last results

Daimakaimura committed:
Scaling the Waku Protocol: A Performance Analysis with Wakurtosis

fryorcraken reviewed (commented):
Would it be worth adding some key figures to the post?

For example:

> The decreased efficiency in data transmission and reception suggests that Discv5’s overhead significantly affects the protocol's performance.

On a 1000 nodes network with X messages per second of size K , the activation of discv5 increased memory consumpion in aveerage per node from X to Y.

Daimakaimura committed:
Added images

Daimakaimura committed:
Updated article with image links

Daimakaimura committed:
Added semantic line breaks

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -13,42 +13,57 @@ categories: waku, wakurtosis
 
-### Introduction
+## Introduction
+The scalability and performance of the Waku protocol are subjects of critical importance to our team. To explore these facets in depth --- i.e. high granularity --- we turned to Wakurtosis, a bespoke simulation framework developed by the VAC-DST team. By studying various network sizes, message rates, and peer discovery setups we aimed to better understand the protocol's capabilities and limitatio ...
```
you have "our team" and "VAC-DST team" in the same paragraph. IMO it sounds weird as I expected "our team" and "VAC-DST" to be the same. Or Am I mistaken?

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+## Introduction
+The scalability and performance of the Waku protocol are subjects of critical importance to our team. 
+To explore these facets in depth --- i.e. high granularity --- we turned to Wakurtosis, a bespoke simulation framework developed by the VAC-DST team. 
+By studying various network sizes, message rates, and peer discovery setups we aimed to better understand the protocol's capabilities and limitations, hence aspects that could benefit from further.
```
```suggestion
By studying various network sizes, message rates, and peer discovery setups we aimed to better understand the protocol's capabilities and limitations, hence aspects that could benefit from further <missing word>.
```

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+Still, there was evidence of increased memory usage with higher message rates, indicating potential areas for optimization. Bandwidth utilization exhibited minor fluctuations as the network size increased, and efficiency in bandwidth usage improved with an increase in the message rate. 
+Some inefficiencies were noted in larger networks, suggesting room for improvement but overall Waku displayed very stable behaviour in terms of memoery and bandwidth usage.
+
+![Bandidth effiencies multipliers over pure payloads in transmission](/static/img/wakurtosis_waku/waku_bandwidth_rx_efficiency.png)
```
```suggestion
![Bandwidth efficiencies multipliers over pure payloads in transmission](/static/img/wakurtosis_waku/waku_bandwidth_rx_efficiency.png)
```

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+![Bandidth effiencies multipliers over pure payloads in transmission](/static/img/wakurtosis_waku/waku_bandwidth_rx_efficiency.png)
+![Bandwidth efficiencies multipliers over pure payloads in reception](/static/img/wakurtosis_waku/waku_bandwidth_tx_efficiency.png)
+
+#### Impact of Discovery v5
```
Looks like this whole section is commented out.

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+Wakurtosis ensures the accuracy of its data by leveraging multiple sources for hardware metrics:
+
+##### Cadvisor (a Google tool)
+Cadvisor providess detailed metrics on resource usage and performance characteristics of Docker containers. 
```
```suggestion
Cadvisor provides detailed metrics on resource usage and performance characteristics of Docker containers. 
```

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+### Scalability
+To overcome scalability challenges, Wakurtosis employs a multi-node approach, running several nodes within one container. 
+This method supports simulations with over 1,000 nodes on a single machine. 
+However, this can introduce unforeseen network effects, potentially affecting some metrics. 
```
Is there a one sentence example of one of these effects?

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+We examined the behaviours of the Waku protocol with multiple network sizes, message rates and peer discovery strategies. 
+Our analysis under baseline conditions --- static network with average degrees of *K=3* and *K=13* --- revealed promising scalability and stability from a memory standpoint. 
+Still, there was evidence of increased memory usage with higher message rates, indicating potential areas for optimization. Bandwidth utilization exhibited minor fluctuations as the network size increased, and efficiency in bandwidth usage improved with an increase in the message rate. 
+Some inefficiencies were noted in larger networks, suggesting room for improvement but overall Waku displayed very stable behaviour in terms of memoery and bandwidth usage.
```
```suggestion
Some inefficiencies were noted in larger networks, suggesting room for improvement but overall Waku displayed very stable behaviour in terms of memory and bandwidth usage.
```

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+Despite certain scalability challenges, Wakurtosis has proved effective in simulating Waku with over 1,000 nodes on a single machine. 
+We examined the behaviours of the Waku protocol with multiple network sizes, message rates and peer discovery strategies. 
+Our analysis under baseline conditions --- static network with average degrees of *K=3* and *K=13* --- revealed promising scalability and stability from a memory standpoint. 
+Still, there was evidence of increased memory usage with higher message rates, indicating potential areas for optimization. Bandwidth utilization exhibited minor fluctuations as the network size increased, and efficiency in bandwidth usage improved with an increase in the message rate. 
```
"bandwidth usage improved with an increase..."

I think this is simply an effect of hardware constraints, so perhaps need to be clarified or left out. Just in case people believe we're capable of miracles. :)

Daimakaimura replied:
LOL ... indeed!

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,99 @@
+![Bandwidth efficiencies multipliers over pure payloads in reception](/static/img/wakurtosis_waku/waku_bandwidth_tx_efficiency.png)
+
+#### Impact of Discovery v5
+<!--When incorporating Discovery v5 into the Waku protocol, we initially observed marked differences in performance metrics, particularly concerning memory consumption and bandwidth usage. However, after adjusting our non-Discv5 simulation parameters to better mirror the natural conditions of Discv5 (namely, by focusing on an average degree closer to 15), these disparities became less pronounced. ...
```
The framing of discv5 here (and previously) can perhaps be a bit clearer. It's not really "integrated" into the protocol or some inherent feature - it's simply another protocol running alongside Waku that helps a Waku node discover and make connections to other peers in the network.

Daimakaimura replied:
Apologies. I fixed it to reflect the fact that it is a separate protocol, ie an addition to the setup.

jm-clius reviewed (commented):
Agree with the comments of other reviewers and added some of my own.

Daimakaimura committed:
Updated heatmaps with K=50 simulations

Daimakaimura committed:
Re-written discussion and  addressed reviewer comments

Daimakaimura committed:
Minor

Daimakaimura commented:
@jm-clius @fryorcraken  thank you so much for your review!. I pushed some changes including addressing your comments. I also finished the Discv5 vs Non-Discv5 discussion with the latest data. Apologies if I forgot something.

fryorcraken commented:
I'll let final review and approval to be handled with Vac team :)

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,85 @@
+When examining the effects on bandwidth utilization, we see varying results.
+For reception rates, with lower message rates we observe that Discv5 consistently utilizes more bandwidth than the corresponding non-Discv5 case.
+However, those differences become less pronounced with higher message rates.
+For transmission rates, both cases show similar performance and both exhibit huge improvements in transmission efficiency at higher message loads.
```
One more note, to reiterate: we wouldn't expect to see an efficiency that's higher than around 1/6. Where this is the case, we should assume there might be losses due to infrastructure (or other) limits.

jm-clius reviewed (approved):
Approving for addressing previous comments. Would leave to Vac to review for framing and other context. Not sure of the intended audience and what we assume they already know, but I would suggest that some concepts would need more contextualisation before being introduced. For example, the basic idea of what discv5 is and does would help readers intuitively understand results a bit a better. Currently the first you read about it is "non-Discv5 simulations", which I wouldn't understand if I didn't have the necessary background already.

kaiserd reviewed (commented):
Thank you :pray: for this post! Feedback inline.
It would be good to also add the 0 Msgs run, suggested by @jm-clius in Discord, to this rlog post.

Daimakaimura committed:
Added relevant links

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,85 @@
+
+## Understanding Wakurtosis
+Wakurtosis is a robust simulation framework which integrates Docker and Kurtosis to create a simulation environment which allows to run highly granular, large scale simulations with a variety of traffic and network patterns. 
+At the core of Wakurtosis is Kurtosis — an orchestration tool responsible for managing containers, known as services, within isolated environments called enclaves. 
```
Added relevant links in throughout the document

Daimakaimura committed:
Added short intro to Waku

Daimakaimura committed:
Edited Waku intro

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,85 @@
+The results presented in here are 1-to-1 simulations.
+
+## Examining the Waku Protocol
+
```
Added to the introduction with links.

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,85 @@
+In simulations with Discv5, we set the *max_peers* parameter to *12* and *50* to approximate similar average degrees. 
+To stress test message throughput, we simulated message rates of *1*, *10*, and *100* messages per second. 
+This combination of network sizes, topologies, message rates, and hardware configurations enabled us to comprehensively evaluate Waku's performance and scalability boundaries under diverse conditions.
+
```
Added the following to clarify this point:

> It's important to note that the Non-Discv5 scenarios presented in this study serve as a theoretical baseline for comparison and are not meant to represent real-world conditions. In a live environment, some form of peer discovery mechanism would be necessary for the functioning of the network. Mechanisms like 'rendezvous' would also introduce additional bandwidth costs. Therefore, the intention behind including Non-Discv5 simulations is not to compare their performance directly with Discv5 but rather to establish a fundamental baseline against which the added complexities and scalling costs of employing a discovery mechanism like Discv5 can be better understood.

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,85 @@
+## Conclusion
+This study underscores the Waku protocol’s resilience and scalability across varied conditions, but also highlights the limitations of Wakurtosis and the need for a more robust simulation infrastructure for demanding scenarios. 
+The protocol’s robustness, evidenced by the absence of message loss, notable stability across network sizes and traffic loads, is a notable takeaway. 
+The addition of Discv5 generally leads to higher memory and bandwidth usage throughout the majority of scenarios. 
```
Slightly tuned the paragraph and also added the discv5 / non-discv-v5 clarification in the simulation setup section

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,85 @@
+Our analysis under baseline conditions --- static network with average degrees of *K=3*, *K=13*, and *K=50* --- revealed good scalability and stability from a memory standpoint. 
+However, there was evidence of increased memory usage with higher message rates, indicating potential areas for optimization. 
+Bandwidth utilization exhibited minor fluctuations as the network size increased, and efficiency in bandwidth usage improved with an increase in the message rate. 
+Some inefficiencies were noted in larger networks, suggesting room for improvement but overall Waku displayed very stable behaviour in terms of memory and bandwidth usage.
```
Added some number and general trends regarding memory usage for both baseline runs and discv5 runs.

Commit:
Expanded intro, links, memory usage expansion and other comments

Daimakaimura commented:
Just pushed some more changes addressing the comments that were pending. Still waiting for the last round of simulations with a rate of 0 msg/s to expand on that. I will ask for re-review once I added those.

Daimakaimura committed:
Added efficiency plots

Daimakaimura committed:
Added new efficiency calculatation and non-load results to discussion

kaiserd reviewed (commented):
Thank you :). Left some comments in-line, will give feedback on the rest later.

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: kaiserd <1684595+kaiserd@users.noreply.github.com>

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: kaiserd <1684595+kaiserd@users.noreply.github.com>

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: kaiserd <1684595+kaiserd@users.noreply.github.com>

Daimakaimura committed:
Added redevous reference

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,146 @@
+[Waku](https://waku.org/) is a family of P2P protocols enabling private, metadata-resistant messaging for Web3 by providing censorship resistance, adaptability, modular design, and a shared service network. Waku is designed to enable communication between decentralized applications (dApps) in a peer-to-peer manner.  
+It serves as an improvement and successor to Ethereum's Whisper protocol, offering better scalability and efficiency.
+
+[Waku Relay](https://rfc.vac.dev/spec/11/), on the other hand, is a specific component or sub-protocol within the broader Waku framework.  
```
Fixed

Daimakaimura committed:
Waku relay as core component in intro

AlbertoSoutullo reviewed (approved):
Didn't see anything to change, specially regarding the technical part. 

When you talk about Kurtosis limitations, remember that I wrote them in Notion if you need it, but I guess this is not the place to talk about them.

Daimakaimura committed:
Moving image folder into /img

Daimakaimura committed:
Added bandwidth plots

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,146 @@
+
+In these plots, efficiency is determined by contrasting the actual bytes transmitted (or received) due to messages with the total bandwidth consumed. 
+
+It's essential to highlight that for better accuracy, we consider the actual message counts that were successfully injected into the network since due to hardware limitations, we've observed that not all anticipated messages get injected, which can skew the interpretation of bandwidth usage. 
```
Switched plots to total bandwidth usage.

Daimakaimura committed:
Fix naming

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,146 @@
+
+To evaluate Waku under varied conditions, we conducted simulations across a range of network sizes, topologies, and message rates. Each simulation lasted 3 hours to reach a steady state.
+
+The network sizes explored included 75, 150, 300, and 600 nodes. For Non-Discv5 simulations, we used static topologies with average node degrees of K=50. In simulations with Discv5, we set the max_peers parameter to 50 to approximate similar average degrees.
```
There is an explanation in the subsection *Clarification on Non-Discv5 Scenarios* bellow. Also, Discv5 is briefly explained in the intro.

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,146 @@
+
+To stress test message throughput, we simulated message rates of 1, and 10 messages per second. We initally run simulations up to a 100 msg/s but we found out the results were unreliable due to simulation hardware limitations and therefore decided not to include them in this analysis.
+
+We also included simulations batches with no load &mdash; i.e. 0 Msg/s. &mdash; to provide a clearer picture of Waku's baseline resource demands and inherent overhead costs stemming from core protocol operations.
```
Not sure if I follow, can you please elaborate a bit?

Daimakaimura committed:
Added memory usage plots

Daimakaimura committed:
Changes from review comments

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,146 @@
+Finally, [Discv5 (Discovery v5)](https://github.com/ethereum/devp2p/blob/master/discv5/discv5.md) is a peer-to-peer networking protocol designed to facilitate node discovery and routing in Ethereum-based decentralized networks. It serves as an upgrade to earlier discovery protocols and is designed to be modular and extensible, allowing it to support various types of decentralized systems beyond j ...
+The protocol enables nodes to find each other and maintain a distributed hash table for efficient routing.
+
+The scalability and performance of the Waku protocol are of critical importance. To explore these facets with high granularity across a wide range of scenarios, we turned to [Wakurtosis](https://github.com/vacp2p/wakurtosis), a bespoke simulation framework developed internally.  
```
Done "Unfortunately, Wakurtosis did not fulfill all our requirements. We will delve into the specifics of these shortcomings in a retrospective article coming shortly"

Daimakaimura commented:
Just pushed the remaining changes. Also included memory vs nodes plots for clarity in the discussion of the results and to better showcase the different scaling of memory. @kaiserd please let me know if there is something I ve missed.

kaiserd reviewed (commented):
Here is the rest of my feedback.

Generally I'd remove the runs where we see reduced  bandwidth use to hardware limitations preventing messages from being injected. 
Seeing bandwidth consumption drop with larger network sizes is misleading.
That is not a real world scenario but just due to testing environment artifacts.

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,201 @@
+
+With no load, memory usage remained consistent around 20-21 MB across all network scales. 
+Under 1 message per second, average memory usage increased slightly to 21-22 MB. 
+The highest memory usage of 23-25 MB was seen with 10 messages per second, especially at 150 nodes. 
```
No, I do not have an explanation for this memory peak. Also if you check the scale in the y-aixs (+/- 4MB) you'll see that despite it shows a peak it actually not that big if you zoom out slightly. But looks to me like sampling noise.

Daimakaimura commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,201 @@
+For Non-Discv5 simulations, we used static topologies with average node degrees of K=50. 
+In simulations with Discv5, we set the max_peers parameter to 50 to approximate similar average degrees.
+
+To stress test message throughput, we simulated message rates of 1, and 10 messages per second. 
```
Added "The message rates rates were always constant and independent of the network size." for clarity.

jm-clius commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,201 @@
+
+With the discovery mechanism, we again see interesting trends in resource usage across different network sizes and traffic loads. 
+
+Transmission bandwidth remainded fairly stable with more nodes.
```
What does this mean? Does this tell us something about Waku messaging protocol, our chosen discovery technique and network size?

jm-clius reviewed (commented):
Thanks for this! I have added some comments, most of which is related to making sure this post is more oriented towards conclusions/interpretation of "Scaling the Waku Protocol" rather than stating simulation observations. In other words, do the tests tell us something about the scalability of the Waku Protocol (e.g. what is our _interpretation_ for each scenario about how Waku Protocols respond to changes in network size and message rates?). These are mentioned in the concluding section, but I think this should be the focus throughout.

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: Hanno Cornelius <68783915+jm-clius@users.noreply.github.com>

AlbertoSoutullo commented:
> @Daimakaimura @AlbertoSoutullo What is the status of the simulations?

I finished them on Thursday around 13:00pm and notified Jordi. He will probably answer about the analysis part when he reads this.

fryorcraken commented:
@Daimakaimura could you ping me and @jm-clius once you have updated the post with the latest data please?

Daimakaimura commented:
> @Daimakaimura could you ping me and @jm-clius once you have updated the post with the latest data please?

Of course!

Daimakaimura committed:
Added new plots and discussion with sub 1 msg/s simulations

Daimakaimura committed:
Merge branch 'DST-Wakurtosis' of https://github.com/vacp2p/vac.dev into DST-Wakurtosis

Daimakaimura commented:
@fryorcraken @kaiserd @jm-clius just pushed the last changes with the results of the last round of simulations. Going to ask for re-review.

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,174 @@
+
+## Introduction
+
+[Waku](https://waku.org/) is a family of P2P protocols enabling private, metadata-resistant messaging for Web3 by providing censorship resistance, adaptability, modular design, and a shared service network. 
```
```suggestion
[Waku](https://waku.org/) is a family of P2P protocols enabling private, scalable messaging for Web3 by providing censorship resistance, adaptability, modular design, and a shared service network. 
```

Not sure what _metadata-resistant_ means. Do you mean _resistant to metadata harvesting_?

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,174 @@
+It serves as an upgrade to earlier discovery protocols and is designed to be modular and extensible, allowing it to support various types of decentralized systems beyond just Ethereum.  
+The protocol enables nodes to find each other, maintaining a distributed hash table.
+
+The scalability and performance of the Waku protocol are of critical importance. 
```
```suggestion
The scalability and performance of the Waku protocol are of critical importance for any projects using Waku and intending to onboard thousands, if not millions, of users. 
```

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,174 @@
+### Scalability  
+
+To overcome scalability challenges, Wakurtosis employs a multi-node approach, running several nodes within one container.
+This method supports simulations with over 1,000 nodes on a single machine.  
```
The biggest network size mentioned below is 600 nodes. Why not using the full capacity of Wakurtosis?

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,174 @@
+
+After this initial decrease, the bandwidth plateaus without significant growth even as rates rise. This contrasts sharply with the 600-node setup, where even without messaging, substantial bandwidth is consumed, comparable to active messaging in smaller networks.
+
+![Total mean bandwidth usage without discovery mechanism](/static/img/wakurtosis_waku/non_discv5_bandwidth.png) 
```
I am not sure I understand the scale. The graph indicates _MB_ for the bandwidth

Is that the total bandwidth used over the 3 hours of message injection?

Would be good to provide some basis. For example, what is the baseline data volume we are injecting? 3 hours * 1 msg per second * msg size.

Also the graph has 5 lines but we stated 4 message rates (0.25, 0.5, 0.75, 1 messages per second).

Daimakaimura replied:
Yes, this is the average total bandwidth usage per node over the 3h hour simulation. I will add the pure payload lines into the plot to show how much data we are actually injecting into the network.

Daimakaimura replied:
@fryorcraken it's 5 series: 0.0, 0.25, 0.5, 0.75, and 1.0 msg/s. I just realised there is a missing legend. I ll fix that too.

fryorcraken commented on rlog/2023-08-04-wakurtosis-waku-scallability.mdx:
```diff
@@ -0,0 +1,174 @@
+
+#### Results with discovery mechanism (Discv5)
+
+With the discovery mechanism discv5, distinct resource usage patterns emerge. Transmission bandwidth remains relatively stable despite an eight-fold node increase, evidencing the protocol's transmission efficiency under discovery. However, reception bandwidth shows a different trend. From 75 to 300 nodes, a significant increase occurs, especially at 0 msg/s (95.83 to 254.7 Mb/s). The 300 to 600 n ...
```
> with bandwidth surging to 497.47 Mb/s without messaging.

This seems worrying. Cc @jm-clius @alrevuelta

alrevuelta replied:
497.47 Mb/s with discv5 and without messaging is indeed worring, but I have never seen this with https://github.com/waku-org/waku-simulator. How can we reproduce it?

AlbertoSoutullo replied:
> 497.47 Mb/s with discv5 and without messaging is indeed worring, but I have never seen this with https://github.com/waku-org/waku-simulator. How can we reproduce it?

If you want we can select one day to do a quick call and reproduce this. It was done with Wakurtosis. You can do it with 3~ commands. But the script to analyze the data is owned by @Daimakaimura. Also @Daimakaimura, are those 497.47 Mb/s constant over the entire simulation? Did you check when this happened? Because if it is a "constant" behaviour, we can test in in 5~ minutes run and will be easier to check.

Daimakaimura replied:
@AlbertoSoutullo those ~500 MB are the average total bandwidth usage per node over the whole simulation (i.e. 3h). @alrevuelta we were also surprised with that result so we re-run the simulation and got similar results, plot attached. We should indeed arrange a meeting and discuss about this because it is indeed weird. Also, keep in mind this also happens without DIscv5.
[analysis_hw.pdf](https://github.com/vacp2p/vac.dev/files/13067256/analysis_hw.pdf)
[analysis_hw.pdf](https://github.com/vacp2p/vac.dev/files/13067271/analysis_hw.pdf)

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: fryorcraken <110212804+fryorcraken@users.noreply.github.com>

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: fryorcraken <110212804+fryorcraken@users.noreply.github.com>

Daimakaimura committed:
Update rlog/2023-08-04-wakurtosis-waku-scallability.mdx

Co-authored-by: fryorcraken <110212804+fryorcraken@users.noreply.github.com>

Daimakaimura committed:
Updated bandiwdth plots with payloads

Daimakaimura committed:
Fixed legend in bandwidth plots

alrevuelta commented:
> @alrevuelta What was the highest node count you tested in waku-simulator? How many peer connections did nodes have in your experiments?

With waku-simulator around 500 nodes, with shadow 1000. Around 50 connections per node.

jm-clius reviewed (commented):
@kaiserd not sure what the plan with this PR is? Doing a bit of github cleanup. :)
